docs/contribution.md

```diff
@@ -143,12 +163,94 @@ DATA: list[dict[str, Any]] = [
         "eos_data": [{"upTime": 665.15, "loadAvg": [0.13, 0.12, 0.09], "users": 1, "currentTime": 1683186659.139859}],
         "inputs": {"minimum": 666},
         # If the test returns messages, it needs to be expected otherwise test will fail.
-        # NB: expected messages only needs to be included in messages returned by the test. Exact match is not required.
+        # The expected message can be a substring of the actual message.
         "expected": {"result": "failure", "messages": ["Device uptime is 665.15 seconds"]},
     },
 ]
 ```
 
+Test example for `anta.tests.connectivity.VerifyReachability` AntaTest that contains atomic results.
+
+``` python
+from anta.tests.connectivity import VerifyReachability
+from tests.units.anta_tests import test
+
+DATA: list[AntaUnitTest] = [
+
+    {
+        "name": "failure-ip",
+        "test": VerifyReachability,
+        "inputs": {"hosts": [{"destination": "10.0.0.11", "source": "10.0.0.5"}, {"destination": "10.0.0.2", "source": "10.0.0.5"}]},
+        "eos_data": [
+            {
+                "messages": [
+                    """ping: sendmsg: Network is unreachable
+                ping: sendmsg: Network is unreachable
+                PING 10.0.0.11 (10.0.0.11) from 10.0.0.5 : 72(100) bytes of data.
+
+                --- 10.0.0.11 ping statistics ---
+                2 packets transmitted, 0 received, 100% packet loss, time 10ms
+
+
+                """,
+                ],
+            },
+            {
+                "messages": [
+                    """PING 10.0.0.2 (10.0.0.2) from 10.0.0.5 : 72(100) bytes of data.
+                80 bytes from 10.0.0.2: icmp_seq=1 ttl=64 time=0.247 ms
+                80 bytes from 10.0.0.2: icmp_seq=2 ttl=64 time=0.072 ms
+
+                --- 10.0.0.2 ping statistics ---
+                2 packets transmitted, 2 received, 0% packet loss, time 0ms
+                rtt min/avg/max/mdev = 0.072/0.159/0.247/0.088 ms, ipg/ewma 0.370/0.225 ms
+
+                """,
+                ],
+            },
+        ],
+        "expected": {
+            "result": "failure",
+            "messages": ["Unreachable Destination 10.0.0.11 from 10.0.0.5 in VRF default"],
+            # This test has implemented atomic results.
+            # Expected atomic results must be specified or the test will fail.
+            "atomic_results": [
+                {
+                    # Expected atomic result description
+                    "description": "Destination 10.0.0.11 from 10.0.0.5 in VRF default",
+                    # If the atomic result is tied to a subset of the test inputs, it needs to be expected here.
```

Comment:
```suggestion
                    # If the atomic result is tied to a subset of the test inputs, it needs to be added here otherwise the test will fail.
```